R/is_file_accessible.R

```diff
@@ -0,0 +1,28 @@
+#' Validate a vector of filenames or filepaths
+#'
+#' This function validates that a vector of filenames or filepaths are
+#' accessible files that: are a file, exist, have read access, and are not
+#' empty. Dropbox files that are visible but not downloaded locally will be
+#' empty files, and they will not pass this validation.
+#'
+#' @param filepaths A character vector
+#'
+#' @return A logical vector the same length as `filepaths`.
+#' @export
+is_file_accessible <- function(filepaths) {
+  if (is.data.frame(filepaths) && identical(length(filepaths), 1L)) {
+    filepaths <- filepaths[[1L]]
+  }
```

Comment:
Just curious, why do we expect/accept a dataframe input and not only vectors?

Reply:
Because tidyverse decided to use a different behavior than base R (for valid reasons that I agree with) when a tibble is subset using `[`.... base R by default "simplifies" to a vector if a data frame is subset to only one column, so it has become very common for users to provide as an argument to a function that is expecting a vector something like `mydata[1]`... but if `mydata` is a tibble, either intentionally or not, then `mydata[1]` will be a data frame with one column. Often users don't even know that their data frame has become a tibble by passing it through a dplyr chain or other common processes. I have dealt with countless questions and support requests related to this over the years, so now I habitually coerce one column data frames to a vector if a vector is what was expected.